src/Pact/Native.hs

```diff
@@ -1632,15 +1641,41 @@ keccak256Def = defGasRNative
 
 hyperlaneDefs :: NativeModule
 hyperlaneDefs = ("Hyperlane",)
-  [ hyperlaneMessageIdDef
-  , hyperlaneDecodeTokenMessageDef
+  [ hyperlaneMessageIdDef TyAny
+  , hyperlaneDecodeTokenMessageDef TyAny
+  ]
+
+hyperlaneAmendedDefs :: NativeModule
+hyperlaneAmendedDefs = ("Hyperlane",)
+  [ hyperlaneMessageIdDef (TyUser (snd hyperlaneDataSchema))
+  , hyperlaneDecodeTokenMessageDef (TyUser (snd hyperlaneDataSchema))
+  , tokenMessageERC20Schema
+  , hyperlaneDataSchema
   ]
 
-hyperlaneMessageIdDef :: NativeDef
-hyperlaneMessageIdDef = defGasRNative
+tokenMessageERC20Schema :: NativeDef
+tokenMessageERC20Schema = defSchema "hyperlane-token-erc20"
+  "Schema type for ERC20 TokenMessage"
```

Comment:
What kind of `TokenMessage` are we referring to here? If it's hyperlane specific it needs to mention this in the docs.